asapdiscovery-ml/asapdiscovery/ml/models.py

```diff
@@ -215,6 +215,40 @@ def pull(self, local_dir: Union[Path, str] = None) -> "LocalEnsembleMLModelSpec"
             **self.dict(exclude={"models"}),
         )
 
+    def pull_plot(
+        self, plotname: str, filename: Optional[str] = None, return_as="memory"
+    ):
+        # check all the base urls are the same
+        base_url = self.models[0].base_url
+        if not all([model.base_url == base_url for model in self.models]):
+            raise ValueError("All models in an ensemble must have the same base url")
+        # get plot at baseurl/plotname
+        plot_url = urljoin(base_url, plotname)
+
+        # pull using requests to in memory
+        try:
+            response = requests.get(plot_url)
+            response.raise_for_status()
+        except Exception as e:
+            warnings.warn(
+                f"Failed to download plot from {plot_url}, skipping. Error: {e}"
+            )
+            return None
+        # return as memory or file
+        if return_as == "memory":
+            return response.content
+        elif return_as == "file":
+            # save to file
+            if not filename:
+                filename = plotname
+            with open(filename, "wb") as f:
+                f.write(response.content)
+        elif return_as == "url":
+            return plot_url
+        else:
+            raise ValueError("return_as must be 'memory' or 'file' or 'url'")
+        return plotname
```

Comment:
should this be `filename`?

Reply:
also, could be moved up into the relevant `elif` block since all the other code paths have their own return statements anyway